runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -4121,8 +4144,49 @@ J9::ValuePropagation::innerConstrainAcall(TR::Node *node)
                addGlobalConstraint(node, TR::VPNonNullObject::create(this));
                }
             }
+         else if ((method->getRecognizedMethod() == TR::jdk_internal_value_ValueClass_newArrayInstance) &&
+                  (node->getFirstChild()->getOpCodeValue() == TR::acall))
+            {
+            /*
+             *   n12n   acall  jdk/internal/value/ValueClass.newArrayInstance(Ljdk/internal/value/CheckedType;I)[Ljava/lang/Object;
+             *   n9n      acall  jdk/internal/value/NullRestrictedCheckedType.of(Ljava/lang/Class;)Ljdk/internal/value/NullRestrictedCheckedType;
+             *   n8n        aloadi  <javaLangClassFromClass>
+             *   n7n          loadaddr  SomeValueClass
+             *   n11n     iload  Test.ARRAY_SIZE
+             */
+            bool isGlobal;
+            TR::Node *firstChildAcallNode = node->getFirstChild();
+            constraint = getConstraint(firstChildAcallNode, isGlobal);
+            TR_ResolvedMethod *owningMethod = symRef->getOwningMethod(comp());
+            TR_OpaqueClassBlock *nullRestrictedCheckedTypeClass = fe()->getClassFromSignature("jdk/internal/value/NullRestrictedCheckedType", 44, owningMethod);
+
+            if (constraint &&
+                constraint->isFixedClass() &&
+                nullRestrictedCheckedTypeClass &&
+                (comp()->fej9()->isInstanceOf(constraint->getClass(), nullRestrictedCheckedTypeClass, true, true) == TR_yes) &&
+                (firstChildAcallNode->getSymbol()->getResolvedMethodSymbol()->getRecognizedMethod() == TR::jdk_internal_value_NullRestrictedCheckedType_of))
```

Comment:
I think this test is now redundant

```
                nullRestrictedCheckedTypeClass &&
                (comp()->fej9()->isInstanceOf(constraint->getClass(), nullRestrictedCheckedTypeClass, true, true) == TR_yes)
```

given the added test that the first child is a call to `NullRestrictedCheckType.of()`.

Also, I think a test of `firstChildAcallNode->getSymbol()->isResolvedMethod()` is needed before the call to `getResolvedMethodSymbol()`.

Reply:
Addressed in [cc811af](https://github.com/eclipse-openj9/openj9/compare/6782be9764d21471a7b2d4a16e71519dbe36ceba..cc811af332be4d1aaabc0318586abad687da9eb0)